both/collections/scheduleEvent.js

```diff
@@ -74,6 +52,23 @@ ScheduleEvents.attachSchema({
       afFieldInput: {
         type: "timeOfDayMinutes"
       }
+    },
+    custom: function() {
+      // SimpleSchema has no place to do multi field validations, so we arbitrarily do this here.
+      var obj = validationObject(this, ['nthDay', 'startMinutes', 'durationMinutes', 'competitionId']);
+
+      var compId = obj.competitionId || ScheduleEvents.findOne(obj.id).competitionId; // Makes is work on client and server, for insert and update.
```

Comment:
Yeah, I struggled with phrasing this...

I just want to explain why this grotesque line is there.